zebra-chain/src/orchard/tree.rs

```diff
@@ -292,6 +295,71 @@ pub struct NoteCommitmentTree {
     cached_root: std::sync::RwLock<Option<Root>>,
 }
 
+/// The block height of the final leaf in a level 16 subtree and
+/// the block height of the first leaf of the next subtree
+#[derive(Clone, Debug)]
+pub struct NoteCommitmentSubtree {
+    /// Index of this subtree
+    pub index: u16,
+    /// Start boundary of this subtree, the block height of its last leaf.
+    pub start: Option<block::Height>,
+    /// End boundary of this subtree, the block height of its last leaf.
+    pub end: block::Height,
```

Comment:
There seem to be some inconsistencies in the field names and docs in this struct:
1. "last leaf" in `start`'s docs looks like a copy-paste typo.
2. Either the `start` field name is wrong, or the struct documentation is wrong. If the field is the start of the next subtree, I'd suggest putting it after the end of the current subtree, and renaming both fields to avoid ambiguity.

I'd suggest also making these changes to the start and end structs.

```suggestion
    /// End boundary of the current subtree, the block height of its last leaf.
    pub current_tree_end: block::Height,
    /// Start boundary of the next subtree, the block height of the next subtree's first leaf.
    pub next_tree_start: Option<block::Height>,
```

Reply:
This is not important now that there is no start field.